js/utils/__tests__/musicutils.test.js

```diff
@@ -0,0 +1,376 @@
+    getModeNumbers,
+    getDrumIndex,
+    getDrumName,
+    getDrumSymbol,
+    getFilterTypes
+} = require("../musicutils");
+
+
+describe("musicutils", () => {
+    it("should set and get Octave Ratio", () => {
+        setOctaveRatio(4);
+        const octaveR = getOctaveRatio();
+        expect(octaveR).toBe(4);
+    });
+});
+
+describe("Temperament Functions", () => {
+    it("getTemperamentsList should return the list of temperaments", () => {
+        expect(getTemperamentsList()).toEqual([
+            [_("Equal (12EDO)"), "equal", "equal"],
+            [_("Equal (5EDO)"), "equal5", "equal5"],
+            [_("Equal (7EDO)"), "equal7", "equal7"],
+            [_("Equal (19EDO)"), "equal19", "equal19"],
+            [_("Equal (31EDO)"), "equal31", "equal31"],
+            [_("5-limit Just Intonation"), "just intonation", "just intonation"],
+            [_("Pythagorean (3-limit JI)"), "Pythagorean", "Pythagorean"],
+            [_("Meantone") + " (1/3)", "1/3 comma meantone", "meantone (1/3)"],
+            [_("Meantone") + " (1/4)", "1/4 comma meantone", "meantone (1/4)"],
+            [_("Custom"), "custom", "custom"]
+        ]);
+    });
+
+    describe("getTemperament", () => {
+        it("should return the correct temperament for a valid key", () => {
+            const equalTemperament = getTemperament("equal");
+            expect(equalTemperament).toHaveProperty("perfect 1");
+            expect(equalTemperament).toHaveProperty("minor 2");
+            expect(equalTemperament).toHaveProperty("pitchNumber", 12);
+        });
+
+        it('should return the correct temperament for equal5 key', () => {
+            const equal5Temperament = getTemperament("equal5");
+            expect(equal5Temperament).toHaveProperty("perfect 1");
+            expect(equal5Temperament).toHaveProperty("minor 2");
+            expect(equal5Temperament).toHaveProperty("pitchNumber", 5);
+        });
+
+        it("should return undefined for an invalid key", () => {
+            const invalidTemperament = getTemperament("invalid");
+            expect(invalidTemperament).toBeUndefined();
+        });
+    });
+
+    describe("getTemperamentKeys", () => {
+        it("should return an array with the correct length", () => {
+            const keys = getTemperamentKeys();
+            expect(keys.length).toBe(10); 
+        });
+
+        it('should return an array containing all keys', () => {
+            const keys = getTemperamentKeys();
+            expect(keys).toEqual(
+                expect.arrayContaining([
+                    "equal",
+                    "equal5",
+                    "equal7",
+                    "equal19",
+                    "equal31",
+                    "just intonation",
+                    "Pythagorean",
+                    "1/3 comma meantone",
+                    "1/4 comma meantone",
+                    "custom"
+                ])
+            );
+        });
+    });
+});
+/*   
```

Comment:
Curious why this commented code is included?

Reply:
When I made test functions for the "Temperament Management " block the results were not matched. Somehow the functions are not working. 

![IMG-20241214-WA0000.jpg](https://github.com/user-attachments/assets/a73f3846-1e80-4d7b-a467-1ac946302015)

I did not want to edit musicutil.js so I left this job to @Aman1919

Reply:
ThenpPlease remove the commented code from this PR.

Reply:
@walterbender this shows error while running the test 
<img width="1027" alt="Screenshot 2024-12-14 at 10 12 19 PM" src="https://github.com/user-attachments/assets/a347b249-244a-4d8c-88c5-5d8563e6e80e" />